linera-storage/src/db_storage.rs

```diff
@@ -319,9 +330,77 @@ enum BaseKey {
     NetworkDescription,
 }
 
+impl BaseKey {
+    fn root_key(&self) -> Vec<u8> {
+        match self {
+            BaseKey::ChainState(chain_id) => {
+                let mut key = vec![INDEX_CHAIN_ID];
+                key.extend_from_slice(chain_id.0.as_bytes().as_slice());
+                key
+            }
+            BaseKey::Certificate(hash) => {
+                let mut key = vec![INDEX_CERTIFICATE];
+                key.extend_from_slice(hash.as_bytes().as_slice());
+                key
+            }
+            BaseKey::ConfirmedBlock(hash) => {
+                let mut key = vec![INDEX_CONFIRMED_BLOCK];
+                key.extend_from_slice(hash.as_bytes().as_slice());
+                key
+            }
+            BaseKey::Blob(blob_id) => {
+                let mut key = vec![INDEX_BLOB_ID];
+                key.push(blob_id.blob_type as u8);
+                key.extend_from_slice(blob_id.hash.as_bytes().as_slice());
+                key
+            }
+            BaseKey::BlobState(blob_id) => {
+                let mut key = vec![INDEX_BLOB_STATE];
+                key.push(blob_id.blob_type as u8);
+                key.extend_from_slice(blob_id.hash.as_bytes().as_slice());
+                key
+            }
+            BaseKey::Event(event_id) => event_root_key(&event_id.chain_id, &event_id.stream_id),
+            BaseKey::BlockExporterState(id) => {
+                let mut key = vec![INDEX_BLOCK_EXPORTER_STATE];
+                key.extend_from_slice(&id.to_le_bytes());
+                key
+            }
+            BaseKey::NetworkDescription => {
+                vec![INDEX_NETWORK_DESCRIPTION]
+            }
+        }
+    }
+}
+
+fn event_key(event_id: &EventId) -> Vec<u8> {
+    let mut key = vec![0];
+    key.extend(bcs::to_bytes(&event_id.index).unwrap());
+    key
+}
+
+fn event_root_key(chain_id: &ChainId, stream_id: &StreamId) -> Vec<u8> {
+    let mut key = vec![INDEX_EVENT_ID];
+    key.extend_from_slice(chain_id.0.as_bytes().as_slice());
+    key.extend(bcs::to_bytes(stream_id).unwrap());
+    key
+}
+
+fn is_chain_state(root_key: &[u8]) -> bool {
+    if root_key.is_empty() {
+        return false;
+    }
+    root_key[0] == INDEX_CHAIN_ID
+}
+
 const INDEX_CHAIN_ID: u8 = 0;
```

Comment:
nit: The expected name would be `CHAIN_ID_TAG`.

Reply:
Ok.